src/ethereum/wallet.rs

```diff
@@ -31,45 +29,41 @@ const DERIVATION_PATH: EthereumDerivationPath<wagyu_ethereum::network::Mainnet>
 pub struct Wallet {
     private_key: clarity::PrivateKey,
     geth_client: Client,
-    dai_contract_addr: Address,
-    pub chain_id: ChainId,
+    chain: ethereum::Chain,
 }
 
 impl Wallet {
-    pub async fn new(
-        seed: Seed,
-        url: Url,
-        dai_contract_addr: Address,
-        chain_id: ChainId,
-    ) -> anyhow::Result<Self> {
+    pub async fn new(seed: Seed, url: Url, chain: ethereum::Chain) -> anyhow::Result<Self> {
```

Comment:
This `ethereum::Chain` is specific to `dai` (and defined in `dai`).

I think we are getting the module boundaries wrong here.

If this chain is for ethereum and it should be defined in the `ethereum` module.

I know it has `DAI` characteristics but I think it's fine has `dai` is a submodule of `ethereum`. `ethereum` can bring together `dai` and `ethereum` concepts.

I would try to move the `Chain` definition and `impl` block in `ethereum` and have some logic around the address defined in `dai` and brought in `ethereum`.

Not sure it makes sense, if it does not, let me know and I can give it a whack.

Reply:
I agree with this. I thought about doing it properly like you propose, but it was late and I was tired so I just re-exported it haha. I'll try!